data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -36,46 +40,52 @@ static Results post(String url, String aadAccessToken, String payload) throws Ex
+                    JSONArray resultsColumns = table0.getJSONArray("Columns");
 
-                HashMap<String, Integer> columnNameToIndex = new HashMap<>();
-                HashMap<String, String> columnNameToType = new HashMap<>();
-                for (int i = 0; i < resultsColumns.length(); i++) {
-                    JSONObject column = resultsColumns.getJSONObject(i);
-                    String columnName = column.getString("ColumnName");
-                    columnNameToIndex.put(columnName, i);
-                    columnNameToType.put(columnName, column.getString("DataType"));
-                }
+                    HashMap<String, Integer> columnNameToIndex = new HashMap<>();
+                    HashMap<String, String> columnNameToType = new HashMap<>();
+                    for (int i = 0; i < resultsColumns.length(); i++) {
+                        JSONObject column = resultsColumns.getJSONObject(i);
+                        String columnName = column.getString("ColumnName");
+                        columnNameToIndex.put(columnName, i);
+                        columnNameToType.put(columnName, column.getString("DataType"));
+                    }
 
-                JSONArray resultsRows = table0.getJSONArray("Rows");
-                ArrayList<ArrayList<String>> values = new ArrayList<>();
-                for (int i = 0; i < resultsRows.length(); i++) {
-                    JSONArray row = resultsRows.getJSONArray(i);
-                    ArrayList<String> rowVector = new ArrayList<>();
-                    for (int j = 0; j < row.length(); j++) {
-                        rowVector.add(row.getString(j));
+                    JSONArray resultsRows = table0.getJSONArray("Rows");
+                    ArrayList<ArrayList<String>> values = new ArrayList<>();
+                    for (int i = 0; i < resultsRows.length(); i++) {
+                        JSONArray row = resultsRows.getJSONArray(i);
+                        ArrayList<String> rowVector = new ArrayList<>();
+                        for (int j = 0; j < row.length(); j++) {
+                            rowVector.add(row.getString(j));
+                        }
+                        values.add(rowVector);
                     }
-                    values.add(rowVector);
-                }
 
-                return new Results(columnNameToIndex, columnNameToType, values);
-            } else {
-                throw new WebException(responseContent, response);
+                    return new Results(columnNameToIndex, columnNameToType, values);
+                } else {
+                    throw new DataWebException(responseContent, response);
+                }
             }
+        } catch (DataWebException e) {
+            throw new DataServiceException(url, "Error in post request", e);
+        } catch (JSONException | IOException e) {
+            throw new DataClientException(url, "Error in post request", e);
         }
         return null;
```

Comment:
Do we ever want to return a null? why not throwing here? Do we swallow any exceptions here?

Reply:
in case of empty response, the entity is null, then it makes sense to return null.